src/ui/datepicker/tests/datepicker-spec.js

```diff
@@ -158,4 +158,40 @@ describe('avDatepicker', () => {
     expect(date).toBe('apple');
   });
 
+  it('should not rollover invalid month values', () => {
+
+    tester.$scope.selectedDate = null;
+    $el = tester.compileDirective(fixtures.regular);
+
+    const ngModel = $el.data('$ngModelController');
+    ngModel.$setViewValue('15/04/1945');
+    tester.flush(DEBOUNCE);
+    tester.$scope.$digest();
+
+    const date = tester.$scope.selectedDate;
+
+    expect(date instanceof Date).toBe(true);
+    expect(date.getFullYear()).toBe(1945);
+    expect(date.getMonth()).not.toBe(2);
```

Comment:
It would probably be better to have a positive test than a negative one. Sure, it should not be 2, but what should it be?